### PR TITLE
[ADEN-10500] BingeBot: set s2 in page level targeting

### PR DESCRIPTION
**Pull request by Loczi94**

### CODE REVIEW

File: platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+	ofType,
+	Targeting,
+	utils,
+} from '@wikia/ad-engine';
 import { Inject, Injectable } from '@wikia/dependency-injection';
+import { shareReplay } from 'rxjs/operators';
+import { props } from 'ts-action';
+
+interface NewViewRenderedProps {
+	s2: string;
```
Comment (fraszczakszymon) [inline platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts:17]: `s2` was defined a long time ago and it was meant to be short. `viewType` would be more descriptive as param. What do you think?

File: platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+}
+
+const newViewRendered = globalAction('[BingeBot] new view rendered', props<NewViewRenderedProps>());
```
Comment (fraszczakszymon) [inline platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts:20]: What about `[BingeBot] view rendered`? Is `new` crucial here?

**Review by fraszczakszymon — APPROVED** [review]

File: platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts (hunk 2 above)
Comment (Loczi94) [reply platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts:20]: No, it was just an idea. `[BingeBot] view rendered` is ok.

**Review by fraszczakszymon — APPROVED** [review]

File: platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts [hunk 3]
```diff
@@ -17,6 +36,18 @@ export class BingeBotTargetingSetup implements TargetingSetup {
...
 		context.set('targeting', { ...context.get('targeting'), ...this.getPageLevelTargeting() });
+
+		communicationService.action$
+			.pipe(
+				ofType(newViewRendered),
+				shareReplay(1), // take only the newest value
+			)
+			.subscribe((action) => {
+				context.set('targeting.s2', action.viewType);
+				eventService.emit(events.PAGE_RENDER_EVENT, {
```
Comment (fraszczakszymon) [inline platforms/bingebot/setup/context/targeting/bingebot-targeting.setup.ts:47]: I'm not sure about subscribing to this with shareReplay. It deserves a separate place. Take a look at mobile-wiki and hooks example.

**Review by fraszczakszymon — APPROVED** [review]

**Pull request merged**